narwhal/worker/src/tx_validator.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use types::Batch;
+
+/// Defines the validation procedure for receiving either a new single transaction (from a client)
+/// of a batch of transactions (from another validator). Invalid transactions will not receive
+/// further processing.
+pub trait TxValidator: Clone + Send + Sync + 'static {
+    type Error: Send + Sync + 'static;
```

Comment:
Sure! That's certainly a two-way door at this point.

Reply:
I guess my problem is that in this PR  error type is not used in any way.

If it can be used for logging let's use it for logging then. If it's not used let's remove it and re-introduce when there is usage

Reply:
Error logging is in f9596ce335aa5a8cc8a7666f35a7c75ff574c87a